app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -439,25 +439,21 @@ def parse_base_item(item)
     def parse_image_name(image, image_ref)
       parts = %r{
         \A
-        (?:
-          (?<host>.*?)
-          (?::(?<port>.*?))?
-          /(?=.*/)
-        )?
-        (?<name>.*?)
-        (?::(?<tag>[^:]*?))?
+          (?:(?:(?<host>[^\.:\/]+\.[^\.:\/]+)|(?:(?<host2>[^:\/]+)(?::(?<port>\d+))))\/)?
+          (?<name>(?:[^:\/@]+\/)*[^\/:@]+)
+          (?:(?::(?<tag>.+))|(?:\@(?<shatag>sha256:.+)))?
         \z
       }x.match(image)
```

Comment:
> @zeari is there a more authoritative place that describes what needs to be parsed and how?

@matthewd I wish I could find one because its more like guess work at this point.
Theres a small hint from the syntax of ```docker pull```
```Usage: docker pull [OPTIONS] NAME[:TAG]  | REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG]```

However sometimes kubernetes returned some odd examples with extended paths such as:

``` www.site.com:1234/example/more.withdots/examples:tag ```

so I'm attempting to cover as many types of names that I can.
specifically, the extended path under a registry which I have only seen returned from kubernetes and nowhere else. 

The must have dots rule is just the only way I could think to differentiate between between registries and names with extended paths...